docs/extensions/shoot-maintenance.md

```diff
@@ -0,0 +1,10 @@
+# Shoot maintenance
+Shoots configure a maintenance time window in which Gardener performs maintenance related tasks like infrastructure reconciliation, updating K8s or machine image versions.
+
+## Restart control plane controllers
+It is possible to configure Gardener to restart/delete certain control plane pods during a shoot's maintenance.
+Stale states and caches are refreshed and thus solve potential dead-locks or starving routines.
+
+Gardener, for example, takes this precautionary measure for `Kube-Controller-Manager` pods. 
```

Comment:
Maybe add the more reasoning here - usually, this should not be necessary as controllers becoming hung should not happen in the first place. It's only an action we take to reduce ops effort because we have observed that indeed some controllers may become hung (KCM) - hard to reproduce and, thus, this is only a work-around until the controller is eventually fixed.